src/Firebase.Tests/OfflineTests.cs

```diff
@@ -1,18 +1,255 @@
+              ""dino1"": {
+                ""ds"": {
+                  ""height"" : 1,
+                  ""length"" : 1,
+                  ""weight"": 1
+                }
+              }
+            }";
+
+        const string ResponseData_TwoDinos = @"{
+              ""dino2"": {
+                ""ds"": {
+                  ""height"" : 2,
+                  ""length"" : 2,
+                  ""weight"": 2
+                }
+              },
+              ""dino3"": {
+                ""ds"": {
+                  ""height"" : 3,
+                  ""length"" : 3,
+                  ""weight"": 3
+                }
+              }
+            }";
+
+        [TestMethod]
+        public void PostDinosaurAndMakeSureItsLoadedWhenCreatingANewDbRef()
+        {
+            var dinosaurDB = GetOfflineDinosaurDB();
+            RealtimeDatabase<Dinosaur> dinosaurDB2 = null;
+
+            try
+            {
+                Dinosaur dino = new Dinosaur(5, 20, 30);
+                dinosaurDB.Post(dino);
+
+                dinosaurDB2 = GetOfflineDinosaurDB();
+
+                Assert.AreEqual(1, dinosaurDB.Database.Count);
+                Assert.AreEqual(1, dinosaurDB2.Database.Count);
+            }
+            finally
+            {
+                CleanUp(dinosaurDB, dinosaurDB2);
+            }
+        }
+
+        [TestMethod]
+        public void PutDinosaurAndMakeSureItsIdenticalToTheOneInTheDatabase()
+        {
+            var dinosaurDB = GetOfflineDinosaurDB();
+
+            try
+            {
+                Dinosaur dino = new Dinosaur(5, 20, 30);
+                dinosaurDB.Put("dino1", dino);
+
+                var recoveredDino = dinosaurDB.Database["dino1"].Deserialize<Dinosaur>();
+
+                Assert.AreEqual(dino.Dimensions.Height, recoveredDino?.Dimensions.Height);
+                Assert.AreEqual(dino.Dimensions.Length, recoveredDino?.Dimensions.Length);
+                Assert.AreEqual(dino.Dimensions.Weight, recoveredDino?.Dimensions.Weight);
+            }
+            finally
+            {
+                CleanUp(dinosaurDB);
+            }
+        }
+
+        [TestMethod]
+        public void PostThreeDinosaursAndRetrieveThemAll()
+        {
+            var dinosaurDB = GetOfflineDinosaurDB();
+
+            try
+            {
+                Dinosaur dino1 = new Dinosaur(1, 1, 1);
+                dinosaurDB.Post(dino1);
+                Dinosaur dino2 = new Dinosaur(2, 2, 2);
+                dinosaurDB.Post(dino2);
+                Dinosaur dino3 = new Dinosaur(3, 3, 3);
+                dinosaurDB.Post(dino3);
+
+                IEnumerable<FirebaseObject<Dinosaur>> offlineDinos = dinosaurDB.Once();
+
+                Assert.AreEqual(3, offlineDinos.Count());
+            }
+            finally
+            {
+                CleanUp(dinosaurDB);
+            }
+        }
+
+        // TODO: Figure out why this test passes when run alone, but fails when running as a group.
```

Comment:
I tried debugging this test and it seems non-deterministic. Sometimes it would succeed when run in a group, sometimes it would fail running it alone. Could the order of other tests run be the issue here?